### DISCUSSION TITLE
Mapsui in MAUI Blazor Hybrid

### DISCUSSION THREAD

**Discussion by bryanrcarlson** [Mapsui/Mapsui#2715] [Q&A] [answered]
Hello -- I'm wondering if it's possible to use Mapsui in a MAUI Blazor Hybrid app? I followed the Blazor setup and got an error: "SkiaSharp currently only works on Web Assembly.". Is there a known solution for this?

Thanks!

**Comment by inforithmics** [comment, marked as answer]
https://github.com/mono/SkiaSharp/issues/1194
Skiasharp needs to be supported as a Browser extension then it should work in Blazor Hybrid.

I have to investigate if a Mapsui Blazor extension would be possible.

**Reply by bryanrcarlson** [reply]
Thanks for the explanation and the link. Blazor Hybrid is sorely in need of a powerful map component and Mapsui will probably fill that gap nicely.

**Reply by pauldendulk** [reply]
I kind of expected this to work without any modifications. Technically it is still rendering in the browser. The best way to investigate this is indeed to first check if this is at all possible with SkiaSharp.

**Comment by fl47506** [comment]
I have the same problem. Want to use Mapsui in a Blazor / Maui App within a razor component. I followed the recommended Maui Setup as well as the Blazor setup from the Quickstart-documentation and some variations but nothing worked. Are there any updates to this issue?
Or even a guideline which way of using Mapsui is the right one for a Blazor / Maui Hybrid App?

Thanks.

**Comment by guillaume-rce** [comment]
Hi,
It's also one of my problems... Are there any updates ?
Thanks a lot.

**Comment by pauldendulk** [comment]
This is still open. This would require mostly investigating how Blazor works. 

One interpretation of server side rendering is that you generate the entire map on the server. This would mean you have to send the updated map image to the client on every change. This may be useful in some special cases, but usually this is not what you want.

What we need is a solution where the server side rendering entails putting the MapControl in the page, including the configuration and viewport state. This is what is pushed to the browser, and in the browser the MapControl would render the map image.
